### DISCUSSION TITLE
Cannot expand sub-relation field

### DISCUSSION THREAD

**Discussion by Yasser85** [pocketbase/pocketbase#2650] [Q&A] [answered]
Hi.

Thank you for this great software.
I have 2 collections in my Pocketbase:
1- shelves (id, name, ......)
2- items (has a relation field "shelfID" point to shelves collection)
I'm using Dart SDK.
When I'm trying to expand the "`getFullList`" :
`expand: 'shelfID'`
It works and fetch all fields from relative record in "shelves" for each item.

but when I expand to only one field "name" because I need the name only, it doesn't do any expansion (in result : `expand : {}`)

I removed API Rules from shelves collection and make it accessible for anyone.
I use it like this:
```
final mats = await pb.collection('items').getFullList(
expand: 'shelfID.name');
```
or in query:
```
final mats = await pb.collection('items').getFullList(
query: {'expand': 'matGrId.name'});
```
thanks

**Comment by ganigeorgiev** [comment, marked as answer]
If you want to restrict the `expand` json response you'll have to specify the `fields` param too:

```dart
pb.collection("example").getFullList(
  expand: "shelfID",
  // note that currently you have to list every field you want to fetch
  // (in the future I may reconsider introducing "*" as several users requested it)
  fields: "id,create,update,collectionName,expand.shelfID.name",
);
```

**Reply by Yasser85** [reply]
Oh, I see now.
maybe I messed it in documentation, or it wasn't clear.
Thanks a lot for your answer.
It will be so helpful if you make "*" available.

**Reply by Yasser85** [reply]
So I need to request it like this: `expand: "shelfID",` not like mentioned in documentations `expand: "shelfID.name",`
Then list it in fields for example like this: fields: "id,itemName,expand.shelfID.name",
Now in expand it looks like that"
`{"id":"2zs14t6gvlkf9jp","created":"","updated":"","collectionId":"","collectionName":"","expand":{"shelfID":{"id":"","created":"","updated":"","collectionId":"","collectionName":"","expand":{},"name":"Sh1"}},"name":"Item 1"}`

Is there a way to omit "`"id":"","created":"","updated":"","collectionId":"","collectionName":"","expand":{},`" in the response when I don't need it?
It's not a big deal, but I just wanted to know.

P.S : when I write ``expand: "shelfID.name"` along with "fields", it doesn't work, the expand returned empty.

**Reply by ganigeorgiev** [reply]
I'm not sure I follow. I don't think there is anywhere in the documentation that mentions the use of `shelfID.name`. The dot-notation in the `expand` field is only for relation fields. `name` is not a relation field in the `shelfID` referencing collection.

> Is there a way to omit ...

You are probably just seeing the stringified version of the [`RecordModel` dto](https://github.com/pocketbase/dart-sdk/blob/master/lib/src/dtos/record_model.dart). In reality if you are sending the `fields` parameter with the request only those requested fields will be returned by the PocketBase server.

**Reply by Yasser85** [reply]
> I'm not sure I follow. I don't think there is anywhere in the documentation that mentions the use of `shelfID.name`. The dot-notation in the `expand` field is only for relation fields. `name` is not a relation field in the `shelfID` referencing collection.

You are absolutely right, It's clear now.
When I saw "expand: 'relField1,relField2.subRelField'," I thought it means that I can request specific fields from the related collection.


> You are probably just seeing the stringified version of the [`RecordModel` dto](https://github.com/pocketbase/dart-sdk/blob/master/lib/src/dtos/record_model.dart). In reality if you are sending the `fields` parameter with the request only those requested fields will be returned by the PocketBase server.

Thanks a lot. It's all clear.
